mitodl/mit-open
# Use new LearningPath model for Fields

# What are the relevant tickets?
#103 

# Description (What does it do?)
This PR updates the old FieldChannel model we used in Infinite to use new LearningPath model. There are some additional changes to the serializers that I've noted in the code comments.

# How can this be tested?
1. Review the OpenAPI docs at http://localhost:8063/api/v1/schema/redoc/#tag/fields
    - In particular, note that subfields, lists, and featured_list are typed differently (and properly) and the request/response sections for create/update.
2. Create some learning paths, either via the ui at http://localhost:8063/learningpaths/ or at http://localhost:8063/api/v1learningpaths/. *Must be logged in as an admin.*
3. **Create Fields:** Create some fields at http://localhost:8063/api/v1/fields/
   - You can see some associated docs at http://localhost:8063/api/v1/schema/redoc/#tag/fields/operation/fields_create
   - Check that using non-learning paths for `featured_list` and `lists` gives reasonable errors.
4. **View the field details and Update:** http://localhost:8063/api/v1/fields/<field_name>. In particular, try:
    - assigning re-ordering the lists within a field
    - assigning and re-ordering subfields within a field 

# Additional Context

The associated frontend code from Infinite Corridor has not been updated.
